tests/rkt_notify_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build host coreos src kvm
+
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	sd_dbus "github.com/coreos/go-systemd/dbus"
+	sd_util "github.com/coreos/go-systemd/util"
+	"github.com/coreos/rkt/tests/testutils"
+	"github.com/godbus/dbus"
+)
+
+func TestNotify(t *testing.T) {
+	if !sd_util.IsRunningSystemd() {
+		t.Skip("Systemd is not running on the host.")
+	}
+
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	conn, err := sd_dbus.New()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	imageFile := getInspectImagePath()
+
+	image, err := filepath.Abs(imageFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// we need to add --silent-sigterm so inspect terminates correctly and the
+	// transient service exits successfully so it disappears from the list of
+	// units
+	sleepTime := 3
+	opts := fmt.Sprintf("-- --silent-sigterm --notify --sleep=%d", sleepTime)
```

Comment:
So the sleep time in line 57 and line 94 are not related and don't need to have the same value. I don't think they should use the same variable then.

It seems racy in line 94 if rkt is slow to start the app. Instead of a single sleep + a single ListUnit, you could check the status regularly until you get the expected state, with a timeout. There is an example in [TestRktStop](https://github.com/coreos/rkt/blob/master/tests/rkt_stop_test.go#L98)